crates/matrix-sdk-base/src/client.rs

```diff
@@ -203,7 +207,7 @@ impl BaseClient {
     ///
     /// * `session` - An session that the user already has from a previous login
     ///   call.
-    pub async fn restore_session(&self, session: Session) -> Result<()> {
+    pub async fn restore_session(&self, session: SessionMeta) -> Result<()> {
```

Comment:
the docs don't really fit the description anymore now... and the fn name is also confusing. Maybe we should rename that, too for consistency and clarity? 
```suggestion
    pub async fn set_session_meta(&self, session: SessionMeta) -> Result<()> {
```

Reply:
Sure, I kept the `restore` name because it also restores the OlmMachine but I don't have strong feelings about this either way.

This should also be changed on the `Store`?

Reply:
if it only sets the meta, then we should adapt that pattern everywhere I'd say.